app/Helpers/database/subscription.php

```diff
@@ -281,19 +224,19 @@ function isUserSubscribedToArticleComments(int $articleType, int $articleID, int
         return false;
     }
 
-    return isUserSubscribedTo(
-        $subjectType,
-        $articleID,
-        $userID,
-        "
-            SELECT DISTINCT ua.*
-            FROM
-                Comment AS c
-                LEFT JOIN UserAccounts AS ua ON ua.ID = c.user_id
-            WHERE
-                c.ArticleType = $articleType
-                AND c.ArticleID = $articleID
-                AND c.user_id = $userID
-        "
-    );
+    $explicitSubcription = Subscription::where('subject_type', $subjectType)
+        ->where('subject_id', $articleID)
+        ->where('user_id', $userID)
+        ->first();
+
+    if ($explicitSubcription) {
+        return $explicitSubcription->state == 1;
+    }
+
+    $implicitSubscription = Comment::where('ArticleType', $articleType)
+        ->where('ArticleID', $articleID)
+        ->where('user_id', $userID)
+        ->exists();
+
+    return $implicitSubscription;
```

Comment:
Similar minor refactors as above can make this more concise and not need to fetch a record:

```php
function isUserSubscribedToArticleComments(int $articleType, int $articleID, int $userID): bool
{
    $subjectType = SubscriptionSubjectType::fromArticleType($articleType);

    if ($subjectType === null) {
        return false;
    }

    // existence check only here
    $doesExplicitSubscriptionExist = Subscription::where('subject_type', $subjectType)
        ->where('subject_id', $articleID)
        ->where('user_id', $userID)
        ->where('state', true) // we can use `true` instead of `1` due to `$casts` in the model
        ->exists();

    if ($doesExplicitSubscriptionExist) {
        return true;
    }

    // immediate return without a variable
    return Comment::where('ArticleType', $articleType)
        ->where('ArticleID', $articleID)
        ->where('user_id', $userID)
        ->exists();
}
```

Reply:
Fetching the Subscription is required for the first query as the use can explicitly unsubscribe from something they're implicitly subscribed to. In that case, the record would exist with a state of 0.

I've changed the second query to use exists.